test/fw/ptl/utils/pbs_testsuite.py

```diff
@@ -1168,33 +1168,80 @@ def _revert_pbsconf_server(self, vals_to_set):
                 if restart_pbs:
                     # Restart all
                     server.pi.restart(server.hostname)
-                    if not server.isUp():
-                        self.fail("Server is not up")
-                    if not self.mom.isUp():
-                        self.fail("Mom is not up")
-                    if not self.scheduler.isUp():
-                        self.fail("Scheduler is not up")
-                    if not self.comm.isUp():
-                        self.fail("comm is not up")
-
+                    rv = server.isUp()
+                    if not rv:
+                        self.logger.error('Server '
+                                          + comm.hostname
+                                          + ' is down')
+                        msg = 'Failed to restart server '
+                        msg += server.hostname
+                        self.assertTrue(rv, msg)
+                    if new_pbsconf["PBS_START_MOM"] == "1":
+                        rv = server.moms.values()[0].isUp()
+                        if not rv:
+                            self.logger.error('mom '
+                                              + server.moms.values()[0]
+                                              + ' is down')
+                            msg = 'Failed to restart mom '
+                            msg += server.moms.values()[0]
+                            self.assertTrue(rv, msg)
+                    rv = server.schedulers['default'].isUp()
+                    if not rv:
+                        self.logger.error('Scheduler is down')
+                        msg = 'Failed to restart scheduler '
+                        msg += server.scheduler.hostname
+                        self.assertTrue(rv, msg)
+                    if self.du.is_localhost(server.hostname):
```

Comment:
Instead of this, it might be better to just add 'comms' to the server object, then you'll be able to call isUp() on all server daemons, just like mom and scheduler. You'll just need to add cls.server.comms = cls.comms at the end in PBSTestSuite.init_comms()